sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/hints.scala

```diff
@@ -30,7 +30,8 @@ import org.apache.spark.sql.catalyst.trees.TreePattern.{TreePattern, UNRESOLVED_
 case class UnresolvedHint(name: String, parameters: Seq[Any], child: LogicalPlan)
   extends UnaryNode {
 
-  override lazy val resolved: Boolean = false
+  override lazy val resolved: Boolean = child.resolved
```

Comment:
Let's add some comments to explain the reason: we need it to be resolved so that the analyzer can keep analyzing other plan nodes.

Reply:
To make up for this change, let's add a check in `CheckAnalysis` and fail if the plan still contains `UnresolvedHint` after analysis.

Reply:
It's ok for me to add this check. 
And note that in the build-in analyzer, all of `UnresolvedHint` will be removed by batch `Remove Unresolved Hints`.

Reply:
Yea, it's just for sanity check, to make sure `UnresolvedHint` shouldn't exist after analysis.